Client/Frontend/Settings/NewTabContentSettingsViewController.swift

```diff
@@ -51,5 +51,10 @@ class NewTabContentSettingsViewController: SettingsTableViewController {
 
         return [firstSection, secondSection]
     }
+
+    override func viewDidDisappear(_ animated: Bool) {
```

Comment:
I really didnt want to add a `onChange` closure to every pref so instead I trigger the refresh event on `viewDidDisappear`. This will have some false positives but I think its fine.

Reply:
Should we use `viewWillDisappear` instead?

Reply:
I think either one is fine. Both should be called.